src/site/content/en/blog/aria-poison-or-antidote/index.md

```diff
@@ -0,0 +1,283 @@
+consider our ARIA active item as the focus, which is read aloud or shown on a Braille display. 
+
+A word on the _ancestor_, _parent_, and _descendant_ terminology. The term descendant refers to the
+fact that an item is contained somewhere inside of another. The reverse term is ancestor, which is
+to say an item is contained by ancestors. For the next container up/down, these may use the more
+specific terms parent/child. For example, imagine a document with a paragraph that has a link
+inside. The link's parent is a paragraph, but it also has the document as an ancestor. And in
+reverse, the document may have many paragraph children, each with links. The links are all
+descendants of the grandparent document.
+
+Back to `aria-activedescendant`. By using it to point from the focused menubar to a specific menu
+item,  the screen reader now knows where the user has moved, but nothing else about the object. What
+is this div thing anyway? That's where the role attribute comes in. We use `role="menubar"` on the
+containing element for the entire thing, then we use `role="menu"` on groups of items, and
+`role="menuitem"` on … drumroll … the individual menu items.
+
+And what if the menuitem can lead to a child menu? The user needs to know that right? For a sighted
+user, there might be a little picture of a triangle at the end of the menu, but the screen reader
+doesn't know how to automatically read images, at least at this point. We can add
+`aria-expanded="false"` on each expandable menuitem to indicate that 1) there is something that can
+be expanded, and 2) it currently is not expanded. As an added touch the author should put
+`role="none"` on the img triangle to indicate it's for prettification purposes only. This prevents
+the screen reader from saying anything annoying about the image that would be redundant at best.
+
+## Dealing with bugs
+
+### Keyboard bugs (HTML!)
+
+Although keyboard access is a part of core HTML, authors mess it up all the time, either because
+they don't use keyboard navigation all that much, or because there is much nuance to get right. 
+
+Examples of bugs:
+
++   A checkbox uses spacebar to toggle, but the author forgot to call `preventDefault()`. Now
+    the spacebar will both toggle the checkbox and page down, which is the default browser behavior
+    for spacebar.
++   An ARIA modal dialog wants to trap tab navigation inside of it, and the author forgets to
+    specifically allow <kbd>Control</kbd>+<kbd>Tab</kbd> through to the browser. Now,
+    <kbd>Control</kbd>+<kbd>Tab</kbd> just navigates within their dialog, and doesn't switch tabs
+    in the browser as it should. Ugh.
++   An author creates a selection list, and implements up/down, but does not implement
+    home/end/pageup/pagedown or first letter navigation. Authors should follow known patterns.
+    Check out the [Resources](#resources) section for more info.
```

Comment:
```suggestion
    home/end/pageup/pagedown or first letter navigation. 
    
Authors should follow known patterns. Check out the [Resources](#resources) section for more information.
```

Reply:
The last two lines apply to the whole list so don't attach them to the last item.